service/src/main/java/uk/gov/hmcts/reform/fpl/service/cafcass/api/CafcassApiGuardianService.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.hmcts.reform.fpl.service.cafcass.api;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Guardian;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.service.ccd.CoreCaseDataService;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.commons.lang3.ObjectUtils.isEmpty;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.unwrapElements;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.wrapElementsWithUUIDs;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class CafcassApiGuardianService {
+    private final CoreCaseDataService coreCaseDataService;
+
+    public boolean validateGuardians(List<Guardian> guardianUpdateList) {
+        return guardianUpdateList.stream()
+            .noneMatch(guardian -> isEmpty(guardian.getGuardianName())
+                                   || isEmpty(guardian.getChildren())
+                                   || guardian.getChildren().stream().anyMatch(String::isEmpty));
+    }
+
+    public boolean checkIfAnyGuardianUpdated(CaseData caseData, List<Guardian> guardianUpdateList) {
+        List<Element<Guardian>> existingGuardians = caseData.getGuardians();
```

Comment:
Probably me just not understanding this  - would this not always return true if there are existing guardians and their details haven't changed?

Reply:
This method will always return false (means no changes) if the existing and update guardians lists are identical regardless of their order.

If there are existing guardians and their details haven't changed but the update list add / delete any guardian, the method will still return true (means has changes).

If there is no changes, we won't fire the CCD event.